src/Database/Connection.php

```diff
@@ -727,6 +728,39 @@ protected function _newLogger(StatementInterface $statement)
     }
 
     /**
+     * Checks if the driver supports EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @return bool
+     */
+    public function canExplain($query)
+    {
+        $query = (string)$query;
+
+        return preg_match('/^\s*(SELECT|INSERT|UPDATE|DELETE)\s/i', $query)
+            && $this->_driver->explainSQL($query) !== false;
+    }
+
+    /**
+     * Executes an EXPLAIN statement for the specified query.
+     *
+     * @param string $query The query to be explained
+     * @param array $params list or associative array of params to be interpolated in $query as values
+     * @param array $types list or associative array of types to be used for casting values in query
+     * @return \Cake\Database\StatementInterface
+     */
+    public function explain($query, array $params = [], array $types = [])
```

Comment:
In my implementation, it works on both, as Query has __toString. But I would remove explain() from Connection class. As @lorenzo mentioned, it is not so different from calling execute() method directly.

At first, I wanted to support Sqlserver by executing the following queries.
```sql
SET SHOWPLAN_ALL ON
SELECT * FROM somethings
SET SHOWPLAN_ALL OFF
```
So I added explain() for abstraction. But I couldn't it. I didn't know that I couldn't execute SET SHOWPLAN_ALL statement via PDO.

So I am fine to call execute() directly in DebugKit.